x-pack/plugins/ingest_manager/server/routes/global_interceptors.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { KibanaRequest, LifecycleResponseFactory, OnPreAuthToolkit } from 'kibana/server';
+import { LIMITED_CONCURRENCY_ROUTE_TAG } from '../../common';
+
+class MaxCounter {
+  constructor(private readonly max: number = 1) {}
+  private counter = 0;
+  valueOf() {
+    return this.counter;
+  }
+  increase() {
+    if (this.counter < this.max) {
+      this.counter += 1;
+    }
+  }
+  decrease() {
+    this.counter += 1;
+  }
+  lessThanMax() {
+    return this.counter < this.max;
+  }
+}
+
+function shouldHandleRequest(request: KibanaRequest) {
+  const tags = request.route.options.tags;
+  return tags.includes(LIMITED_CONCURRENCY_ROUTE_TAG);
+}
+
+const LIMITED_CONCURRENCY_MAX_REQUESTS = 250;
+const counter = new MaxCounter(LIMITED_CONCURRENCY_MAX_REQUESTS);
+
+export function preAuthHandler(
+  request: KibanaRequest,
+  response: LifecycleResponseFactory,
+  toolkit: OnPreAuthToolkit
+) {
+  if (!shouldHandleRequest(request)) {
+    return toolkit.next();
+  }
+
+  if (!counter.lessThanMax()) {
+    return response.customError({
+      body: 'Too Many Agents',
+      statusCode: 503,
+      headers: {
+        'Retry-After': '30',
+      },
+    });
+  }
+
+  counter.increase();
+
+  // requests.events.aborted$ has a bug where it's fired even when the request completes...
```

Comment:
how can we decrease the counter if/when the bug is fixed? 🤔

Reply:
There was some discussion about this starting in https://github.com/elastic/kibana/pull/70495#issuecomment-656153624

It's a "bug" but there is a test to ensure it works. There's also a possibility Platform will add a `request.events.completed$ ` which we can use instead

Reply:
> It's a "bug" but there is a test to ensure it works. There's also a possibility Platform will add a request.events.completed$ which we can use instead

This is entirely pedantic, it's not really a bug as much as an "implementation detail" per https://github.com/elastic/kibana/pull/70495#issuecomment-657744810

Reply:
@kobelb agreed. I updated the comment in https://github.com/elastic/kibana/pull/71552/files/8174114aec75a3e8a528fdc8dcc04e1c75adff53#diff-ce1cb1cde75c2cf4e5135e1280d24a9dR64-R65